src/main/java/Sally.java

```diff
@@ -0,0 +1,291 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+public class Sally {
+    protected static ArrayList<Task> list = new ArrayList<>();
+    private static Scanner sc;
+
+    public static void main(String[] args) {
+        printBorder();
+        System.out.println("Hello! I'm Sally");
+        System.out.println("What can I do for you?");
+        printBorder();
+
+        try {
+            readsFile();
+        } catch (FileNotFoundException e) {
+        }
+        sc = new Scanner(System.in);
+        messaging();
+    }
+
+    public static void readsFile() throws FileNotFoundException {
+        String filePath = "D:/NUS/Y2/S1/CS2103T/ip/data/Sally.txt";
+        File file = new File(filePath);
+        Scanner sc = new Scanner(file);
+
+        while(sc.hasNext()) {
+            String input = sc.nextLine(); //scans the first input of file
+            String[] arrOfInput = input.split("\\|");
+
+            //Variables to create new Task
+            String taskTypeString = arrOfInput[0].trim();
+            String isDoneString = arrOfInput[1].trim();
+            String description = arrOfInput[2].trim();
+            String moreInfo = "";
+            if (taskTypeString.equals("E") || taskTypeString.equals("D")) {
+                moreInfo = moreInfo + arrOfInput[3].trim();
+            }
+
+            //Convert String to each variable type
+            Task.Type taskType = toTaskType(taskTypeString);
+            boolean isDone = toIsDone(isDoneString);
+
+            Task.makeTask(description, moreInfo, taskType, false);
+            int maxLength = list.size();
+            Task task = list.get(maxLength - 1);
+
+            if (isDone) {
+                task.markAsDone();
+            } else {
+                task.markAsUndone();
+            }
+        }
+    }
+
+    public static boolean toIsDone(String s) {
+        if (s.contains("1")) {
+            return true;
+        } else if (s.contains("0")) {
+            return false;
+        }
+        return false;
+    }
+
+    public static Task.Type toTaskType(String s) {
+        try {
+            if (s.contains("T")) {
+                return Task.Type.TODO;
+            } else if (s.contains("D")) {
+                return Task.Type.DEADLINE;
+            } else if (s.contains("E")) {
+                return Task.Type.EVENT;
+            } else {
+                throw new SallyException.SallyInvalidInputException();
+            }
+        } catch (SallyException e) {
+            System.out.println(e);
+        }
+        return Task.Type.TODO;
+    }
+
+    public static void savesFile() throws IOException {
+        String filePath = "D:/NUS/Y2/S1/CS2103T/ip/data/Sally.txt";
```

Comment:
Consider storing `filePath` as a class field. This will make changing `filePath` easier as you only need to change it in one place.